opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -81,6 +99,57 @@ namespace Opm {
                 for (size_t globalIndex = 0; globalIndex < multpvData.size(); globalIndex++)
                     values[globalIndex] *= multpvData[globalIndex];
             }
+
+            // deal with the region multiplier for porosity
+            if (deck->hasKeyword("MULTREGP")) {
+                const DeckKeyword& multregpKeyword = deck->getKeyword("MULTREGP");
+                for (int recordIdx = 0; recordIdx < multregpKeyword.size(); ++recordIdx) {
+                    const DeckRecord& multregpRecord = multregpKeyword.getRecord(recordIdx);
+
+                    int regionIdx = multregpRecord.getItem("REGION").template get<int>(0);
+                    std::string regionType = multregpRecord.getItem("REGION_TYPE").template get<std::string>(0);
+                    double multValue = multregpRecord.getItem("MULTIPLIER").template get<double>(0);
+                    boost::to_upper(regionType);
```

Comment:
On that note, I see you've used `boost` for uppercasing the string. It's obviously possible, but it's not necessarily all that fast because it deals with encodings, and we offer this https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/Utility/String.hpp#L10

And yes; it's surely a NIH, but it was really the only thing we used from `boost::algorithm::string`, and it improved compilations time quite a lot.